documentatie/flowcharts/date-entry-forms.md

```diff
@@ -0,0 +1,211 @@
+    green-check-styling([valid styling])
+
+    page-visited{page visited before?}
+    current-page{current page?}
+    submitted{submitted?}
+
+    errors{errors?}
+    warnings{warnings?}
+    empty{empty?}
+
+    %% flow
+    flow-start --> page-visited
+    page-visited -- no --> disabled-styling
+    page-visited -- yes --> current-page
+    current-page -- yes --> current-styling
+    current-page -- no --> submitted
+    submitted -- no --> unsubmitted-styling
+    submitted -- yes --> errors
+    errors -- yes --> error-styling
+    errors -- no --> warnings
+    warnings -- yes --> warning-styling
+    warnings -- no --> empty
+    empty -- yes --> empty-styling
+    empty -- no --> green-check-styling
+
+```
+
+
+## Render form
+
+Render happens based on last received API response.
+
+```mermaid
+flowchart TD
+
+    %% elements
+    flow-start([start])
+
+    render-empty-fields([render with empty fields])
+    render-cached-data([render with cached data])
+    flow-end([end])
+
+    render-submitted-data([render with submitted data])
+
+    show-error([show error])
+    show-warning([show warning])
+    show-checked-accepted(["show checked checkbox
+        warnings accepted"])
+    show-unchecked-accepted(["show unchecked checkbox
+        warnings accepted"])
+
+    page-submitted{page submitted?}
+    error-cur-page{error for current page?}
+    warning-cur-page{warning for current page?}
+    cached-input-available{cached input available?}
+    warning-accepted{"warning(s) accepted?"}
+
+    %% flow
+    flow-start --> page-submitted
+
+    page-submitted -- no --> cached-input-available
+    cached-input-available -- yes --> render-cached-data
+    cached-input-available -- no --> render-empty-fields
+    page-submitted -- yes --> render-submitted-data
+    
+    render-submitted-data --> error-cur-page
+
+    error-cur-page -- yes --> show-error
+    show-error --> flow-end
+    error-cur-page -- no --> warning-cur-page
+
+    warning-cur-page -- yes --> show-warning
+    show-warning --> warning-accepted
+    warning-accepted -- yes --> show-checked-accepted
+    warning-accepted -- no --> show-unchecked-accepted
+    show-checked-accepted --> flow-end
+    show-unchecked-accepted --> flow-end
+    warning-cur-page -- no --> flow-end
+```
+
+
+## Click "Volgende"
+
+```mermaid
+flowchart TD
+
+    %% elements
+    flow-start([start])
+    go-to-prev-page([go to previous page])
+    abort-input([abort input])
+
+    error-any-prev-page{"error for any
+        previous page?"}
+    error-cur-page{"error for 
+        current page?"}
+    warning-cur-page{"warning for
+        current page?"}
+    warnings-accepted{"warning(s) accepted?"}
+    user-addresses-error{user addresses error}
+
+    user-addresses-warning{user addresses warning}
+
+    
+    render-error(render error)
+    render-warning(render warning)
+
+    click-next(click next)
+    call-api(call api)
+    change-input(change input)
```

Comment:
the 'hide accept warnings checkbox' item should be added between this and 'click next'

Reply:
Shouldn't that then go in the flowchart about rendering? I'll add it there.